### PR TITLE
MTDSA-9064 Play 28 Upgrades, Dependencies and Plugins

### PR DESCRIPTION
**Pull request by ngangaNjiraini**

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.5.2","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/home/nganga/Documents/idea-IC-193.6494.35/jbr/bin/java","-Xms100m","-Xmx100m","-classpath","/home/nganga/.IdeaIC2019.3/config/plugins/Scala/launcher/sbt-launch.jar","xsbt.boot.Boot","-bsp","--sbt-launch-jar=/home/nganga/.IdeaIC2019.3/config/plugins/Scala/launcher/sbt-launch.jar"]}
```
Comment (james-work-account) [inline .bsp/sbt.json:1]: remove

File: app/v2/connectors/DesConnector.scala [hunk 2]
```diff
@@ -52,7 +56,7 @@ class DesConnector @Inject()(http: HttpClient,
 
     http.POST[CreateSavingsAccountRequest,
       CreateSavingsAccountConnectorOutcome](url,
-      createSavingsAccountRequestData.createSavingsAccount)(writes, reads[CreateSavingsAccountResponse], desHeaderCarrier, implicitly)
+      createSavingsAccountRequestData.createSavingsAccount)(writes, reads[CreateSavingsAccountResponse], desHeaderCarrier(), implicitly)
```
Comment (james-work-account) [inline app/v2/connectors/DesConnector.scala:59]: needs content-type additional header

File: app/v2/connectors/DesConnector.scala [hunk 3]
```diff
@@ -95,7 +99,7 @@ class DesConnector @Inject()(http: HttpClient,
       DesSavingsInterestAnnualIncome.fromMtd(
         incomeSourceId,
         amendSavingsAccountAnnualSummaryRequest.savingsAccountAnnualSummary))(
-      DesSavingsInterestAnnualIncome.writes, reads[DesAmendSavingsAccountAnnualSummaryResponse], desHeaderCarrier, implicitly)
+      DesSavingsInterestAnnualIncome.writes, reads[DesAmendSavingsAccountAnnualSummaryResponse], desHeaderCarrier(), implicitly)
```
Comment (james-work-account) [inline app/v2/connectors/DesConnector.scala:102]: needs content-type additional header

File: project/plugins.sbt [hunk 4]
```diff
@@ -14,22 +14,17 @@
...
-resolvers += "HMRC Releases" at "https://dl.bintray.com/hmrc/releases"
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "2.12.0")
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.2.0")
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "1.11.0")
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.1.0")
+//resolvers += Resolver.url("hmrc-sbt-plugin-releases", url("https://dl.bintray.com/hmrc/sbt-plugin-releases"))(Resolver.ivyStylePatterns)
```
Comment (james-work-account) [inline project/plugins.sbt:17]: don't forget to remove these

**Review by james-work-account — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -22,23 +22,21 @@ object AppDependencies {
...
+    "uk.gov.hmrc"   %% "play-hmrc-api"             % "6.2.0-play-28",
+    "org.typelevel" %% "cats-core"                 % "2.6.1",
   )
 
   def test(scope: String = "test, it"): Seq[sbt.ModuleID] = Seq(
-    "org.scalatest" %% "scalatest" % "3.2.3" % scope,
-    "com.vladsch.flexmark" % "flexmark-all" % "0.36.8" % scope,
-    "org.scalamock" %% "scalamock" % "5.1.0" % scope,
-    "org.pegdown" % "pegdown" % "1.6.0" % scope,
+    "org.scalatest"          %% "scalatest"          % "3.2.3"             % scope,
```
Comment (DreamcastKnight) [inline project/AppDependencies.scala:31]: scalatest can be 3.2.9

File: project/AppDependencies.scala [hunk 6]
```diff
@@ -22,23 +22,21 @@ object AppDependencies {
...
+    "com.vladsch.flexmark"   % "flexmark-all"        % "0.36.8"            % scope,
+    "org.scalamock"          %% "scalamock"          % "5.1.0"             % scope,
+    "org.pegdown"            % "pegdown"             % "1.6.0"             % scope,
 
-    "com.typesafe.play" %% "play-test" % PlayVersion.current % scope,
+    "com.typesafe.play"      %% "play-test"          % PlayVersion.current % scope,
 
-    "org.scalatestplus.play" %% "scalatestplus-play" % "3.1.3" % scope,
+    "org.scalatestplus.play" %% "scalatestplus-play" % "3.1.3"             % scope,
```
Comment (DreamcastKnight) [inline project/AppDependencies.scala:38]: can be 5.1.0 looking at other deppendency prs

**Review by DreamcastKnight — APPROVED** [review]

**Pull request merged**